desktop/src/com/unciv/app/desktop/UncivServer.kt

```diff
@@ -12,12 +12,31 @@ import java.io.File
 
 
 internal object UncivServer {
+    private var serverPort = 8080
+
     @JvmStatic
-    fun main(arg: Array<String>) {
+    fun main(args: Array<String>) {
+        args.forEach { arg ->
+            when {
+                arg.startsWith("-port=") -> {
+                    try {
```

Comment:
If we already know that it might not be a number, we can use toIntOrNull and get rid of try/catch